### PR TITLE
Refactor linux autobuild

### PR DESCRIPTION
**Pull request by ann0see**
**Short description of changes**
Moves some files around to be consistent with Windows/Android build
Removes unneeded files

Changelog entry might need to be split.

**Context: Fixes an issue?**
Fixes: #2430 

**Does this change need documentation? What needs to be documented and how?**

**Status of this Pull Request**
Finished
**What is missing until this pull request can be merged?**
Needs a second review
## Checklist

- [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
- [x] I tested my code and it does what I want
- [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
- [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
- [x] I've filled all the content above

### CODE REVIEW

File: distributions/debian/control [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
 Priority: optional
-Maintainer: "Marc Landolt jr" <debian@marclandolt.ch>
+Maintainer: "The Jamulus Development Team" <team@jamulus.io>
 Build-Depends:
  debhelper (>= 9),
  libjack-jackd2-dev,
  qtbase5-dev,
  qttools5-dev-tools,
-Standards-Version: 3.9.7
+Standards-Version: é&%JAMVERSION%&è
```
Comment (hoffie) [inline distributions/debian/control:10]: That seems wrong to me. I know that this has been there and it looks like some magic string which someone tried to replace with the Jamulus version, but this is simply wrong. Standards-Version describes the Standard for the control file. This is a (Debian-specific number](https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-standards-version).

File: linux/deploy_deb.sh [hunk 2]
```diff
@@ -21,9 +21,6 @@ do
 done
 echo
 
-echo "Copying modified control file for proper dependencies/maintainer logic"
-cp distributions/autobuilddeb/control debian/control
-
 echo "${VERSION} building..."
 
 sed -i "s/é&%JAMVERSION%&è/${VERSION}/g" debian/control
```
Comment (hoffie) [inline linux/deploy_deb.sh:26]:
```suggestion

```

(See above for reasoning)

File: distributions/debian/control [hunk 3]
```diff
@@ -1,43 +1,43 @@
...
 Homepage: https://jamulus.io
 Vcs-Git: git://github.com/jamulussoftware/jamulus.git
 Vcs-Browser: https://github.com/jamulussoftware/jamulus
 
 Package: jamulus
 Architecture: any
-Depends:
- ${shlibs:Depends},
- ${misc:Depends},
+Depends: libc6 (>= 2.17), libgcc1 (>= 1:3.0), libjack-jackd2-0 (>= 1.9.5~dfsg-14) | libjack-0.116, libqt5core5a (>= 5.5.0), libqt5gui5 (>= 5.0.2) | libqt5gui5-gles (>= 5.0.2), libqt5network5 (>= 5.0.2), libqt5widgets5 (>= 5.2.0), libqt5xml5 (>= 5.0.2), libstdc++6 (>= 5.2)
```
[the remark this answers is not in the document]
Comment (ann0see) [reply distributions/debian/control:17]: Yes, I think so.

[the remark this answers is not in the document]
Comment (ann0see) [reply distributions/debian/control:20]: Done.

**Comment by ann0see** [comment]
Sorry for the development noise here. Will set it to draft.

File: linux/deploy_deb.sh (hunk 2 above)
Comment (ann0see) [reply linux/deploy_deb.sh]: Done.

**Review by hoffie — COMMENTED** [review]
Looks good to me now. Two open questions:
- Flatpak is removed because... there's noone on the team maintaining it / using it to publish updates?
- I think the non-autobuild debian logic removal is fine. Maybe still ping the official package maintainer and ask about it to be sure? (mirabilos)

**Comment by ann0see** [comment]
> Flatpak is removed because... there's noone on the team maintaining it / using it to publish updates?

Exactly. Although I have access to the flatpak repo (you probably also have) it's not official. Therefore I don't think it makes sense having it here and on the repo of flatpak.

**Comment by ann0see** [comment]
> I think the non-autobuild debian logic removal is fine. Maybe still ping the official package maintainer and ask about it to be sure? (mirabilos)

I remember him saying that debian uses its own logic anyways. 

@mirabilos could you please comment if modifying the debian folder does any harm for you? Probably the debian build in their repo is built different anyway.

**Comment by mirabilos** [comment]
ann0see dixit:

***@***.*** could you please comment if modifying the debian folder does
***@***.*** harm for you? Probably the debian build in their repo is built
***@***.*** anyway.

The extraction mechanism for upstream sources in Debian
immediately removes any top-level debian subdirectory,
if extant, so we can add the Debian packaging in that place.

bye,
//mirabilos
-- 
Support mksh as /bin/sh and RoQA dash NOW!
‣ src:bash (406 (433) bugs: 0 RC, 275 (295) I&N, 131 (138) M&W, 0 F&P) + 208
‣ src:dash (91 (106) bugs: 0 RC, 51 (55) I&N, 40 (51) M&W, 0 F&P) + 63 ubu
‣ src:mksh (1 bug: 0 RC, 0 I&N, 1 M&W, 0 F&P)
dash has two RC bugs they just closed because they don’t care about quality…

**Comment by ann0see** [comment]
Ok. So you're basically not dependent on anything this PR is doing.

**Comment by ann0see** [comment]
@hoffie I installed the resulting deb and it seemed to work fine. However the text description in Gnome software seemed wring (but that's not important at all)

**Comment by pljones** [comment]
Can I ask for a quick feature request as part of this?  Can you add a Jamulus Server .desktop file to the Linux .deb?  Or doesn't it fit easily here?

**Comment by hoffie** [comment]
> However the text description in Gnome software seemed wring

In what way?

> Can I ask for a quick feature request as part of this? Can you add a Jamulus Server .desktop file to the Linux .deb? Or doesn't it fit easily here?

I think it'd be good to keep "Refactor" PRs to be exactly that. Refactoring. :)

**Comment by mirabilos** [comment]
Peter L Jones dixit:

>Jamulus Server .desktop file

Why would you even install the headless server if you have
a system that’s using .desktop files in the first place?

Just install the full package there… headless is just a hack
to avoid extra dependencies, which those systems that read
.desktop files would usually have already anyway.

bye,
//mirabilos
-- 
22:59⎜<Vutral> glaub ich termkit is kompliziert | glabe nicht das man
damit schneller arbeitet | reizüberflutung │ wie windows │ alles evil
zuviel bilder │ wie ein spiel | 23:00⎜<Vutral> die meisten raffen auch
nicht mehr von windows | 23:01⎜<Vutral> bilderbücher sind ja auch nich
wirklich verbreitet als erwachsenen literatur	‣ who needs GUIs thus?

**Comment by pljones** [comment]
> Why would you even install the headless server if you have a system that’s using .desktop files in the first place?

Where did I mention headless server?

**Review by hoffie — APPROVED** [review]

**Comment by ann0see** [comment]
> In what way?

The gnome software output didn’t say what is written as description but something else sounding like a description. I don’t know exactly what it said.

**Comment by ann0see** [comment]
Ok. See 

grafik

I assume it's from the Debian repository, not the deb file

**Comment by ann0see** [comment]
> Can I ask for a quick feature request as part of this? Can you add a Jamulus Server .desktop file to the Linux .deb? Or doesn't it fit easily here?

I thought @jujudusud did that once (and somehow it disappeared). Could you please open a feature request on that?

**Comment by hoffie** [comment]
> Ok. See
> 
> grafik
> 
> I assume it's from the Debian repository, not the deb file

This seems to match the man page: https://manpages.debian.org/testing/jamulus/Jamulus.1.en.html
This is part of the tree; https://github.com/jamulussoftware/jamulus/blob/master/distributions/Jamulus.1

> I thought @jujudusud did that once (and somehow it disappeared). Could you please open a feature request on that?

I agree it should be a separate issue. The file is still there, btw, it's just missing from `debian/jamulus.install`. I've got a [test build running](https://github.com/hoffie/jamulus/actions/runs/1935648098). If it works, I'll send a PR, if it doesn't, I'll open the extra issue.

**Comment by ann0see** [comment]
@pljones could you please review this PR? I think we should merge it soon (to enable hoffie starting to work on the CI refactoring)

**Review by pljones — APPROVED** [review]
Looks fine to me.

**Pull request merged**